src/python/gudhi/cubical_complex.pyx

```diff
@@ -31,6 +31,7 @@ cdef extern from "Persistent_cohomology_interface.h" namespace "Gudhi":
     cdef cppclass Cubical_complex_persistence_interface "Gudhi::Persistent_cohomology_interface<Gudhi::Cubical_complex::Cubical_complex_interface<>>":
         Cubical_complex_persistence_interface(Bitmap_cubical_complex_base_interface * st, bool persistence_dim_max)
         vector[pair[int, pair[double, double]]] get_persistence(int homology_coeff_field, double min_persistence)
+        vector[pair[int, pair[pair[double, int], pair[double, int]]]] get_cofaces_of_cubical_persistence_pairs(int homology_coeff_field, double min_persistence)
```

Comment:
For compatibility with numpy and the rest of the python ecosystem, we are very slowly trying to avoid complicated mixed structures, and use homogeneous arrays where possible (`intervals_in_dimension`). For instance, that could mean a numpy array (n,2) of float, a numpy array (n,2) of int, and only then group those 2 arrays in a tuple, get one such tuple per dimension and put them in a list or dict. I am not saying this is **the** right interface, I would just like you to look at the place where you use this code with tensorflow or other, and see if an interface like that could help (I am pretty sure it would with pytorch or jax where I think input[intarray] is the differentiable diagram we are looking for).

Reply:
OK. Now the output is a numpy array (I removed the float outputs)